src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs

```diff
@@ -902,7 +902,7 @@ public bool CanWeightComputationBeAvoided(Float instanceDistanceToBestOldCluster
                     arrDv.AddColumn("Features", PrimitiveType.FromKind(DataKind.R4), clusters);
                     arrDv.AddColumn("Weights", PrimitiveType.FromKind(DataKind.R4), totalWeights);
                     var subDataViewCursorFactory = new FeatureFloatVectorCursor.Factory(
-                        TrainUtils.CreateExamples(arrDv.GetDataView(), null, "Features", weight: "Weights"), CursOpt.Weight | CursOpt.Features);
+                        new RoleMappedData(arrDv.GetDataView(), null, "Features", weight: "Weights"), CursOpt.Weight | CursOpt.Features);
```

Comment:
>Features [](start = 71, length = 8)

DefaultColumnNames?

Reply:
Yup. And I see here instead of `DefaultColumnNames.Weight` which is `"Weight"` we instead used `"Weights"`, which is not great. (Just inconsistent, probably doesn't change things.)

---
In reply to: [200460833](https://github.com/dotnet/machinelearning/pull/496#discussion_r200460833) [](ancestors = 200460833)